graphsync.go

```diff
@@ -196,7 +196,7 @@ type BlockData interface {
 type IncomingRequestHookActions interface {
 	SendExtensionData(ExtensionData)
 	UsePersistenceOption(name string)
-	UseLinkTargetNodeStyleChooser(traversal.LinkTargetNodeStyleChooser)
+	UseLinkTargetNodePrototypeChooser(traversal.LinkTargetNodePrototypeChooser)
```

Comment:
Note that there's a small change to the exported interface of this library.  I applied the rename of `s/NodeStyle/NodePrototype/` universally, for consistency.

There's no behavioral change, so any yet-further-downstream consumers should be able to literally apply that regexp when updating and be fine.